test/firmata.test.js

```diff
@@ -1309,4 +1317,167 @@ describe("board", function() {
+    serialPort.lastWrite[9].should.equal(END_SYSEX);
+    done();
+  });
+
+  it("has a serialStartReading method that sets READ_CONTINUOUS mode", function(done) {
+    board.serialStartReading(0x08);
+    serialPort.lastWrite[2].should.equal(SERIAL_READ | 0x08);
+    serialPort.lastWrite[3].should.equal(0);
+    serialPort.lastWrite[4].should.equal(END_SYSEX);
+    done();
+  });
+
+  it("has a serialStopReading method that sets STOP_READING mode", function(done) {
+    board.serialStopReading(0x08);
+    serialPort.lastWrite[2].should.equal(SERIAL_READ | 0x08);
+    serialPort.lastWrite[3].should.equal(1);
+    serialPort.lastWrite[4].should.equal(END_SYSEX);
+    done();
+  });
+
+  it("has a serialClose method", function(done) {
+    board.serialClose(0x09);
+    serialPort.lastWrite[2].should.equal(SERIAL_CLOSE | 0x09);
+    done();
+  });
+
+  it("has a serialFlush method", function(done) {
+    board.serialFlush(0x02);
+    serialPort.lastWrite[2].should.equal(SERIAL_FLUSH | 0x02);
+    done();
+  });
+
+  it("has a serialListen method that switches software serial port", function(done) {
+    var spy = sinon.spy(serialPort, "write");
+    board.serialListen(0x08);
+    serialPort.lastWrite[2].should.equal(SERIAL_LISTEN | 0x08);
+    serialPort.lastWrite[3].should.equal(END_SYSEX);
+    should.equal(spy.callCount, 1);
+    spy.restore();
+    done();
+  });
+
+  it("should not send a SERIAL_LISTEN message for a hardware serial port", function(done) {
+    var spy = sinon.spy(serialPort, "write");
+    board.serialListen(0x01);
+    should.equal(spy.callCount, 0);
+    spy.restore();
+    done();
+  });
+
+  it("can read a single byte from a serial port", function(done) {
+    var lsb = 242 & 0x7F;
+    var msb = (242 >> 7) & 0x7F;
+
+    serialPort.once("data", function(buffer) {
+      should.deepEqual(buffer, [START_SYSEX, SERIAL_MESSAGE, SERIAL_REPLY | 0x08, lsb, msb, END_SYSEX]);
+      done();
+    });
+
+    serialPort.emit("data", [START_SYSEX, SERIAL_MESSAGE, SERIAL_REPLY | 0x08, lsb, msb, END_SYSEX]);
+  });
+
+  it("can read a byte array from a serial port", function(done) {
+    var inBytes = [
+      242 & 0x7F,
+      (242 >> 7) & 0x7F,
+      243 & 0x7F,
+      (243 >> 7) & 0x7F,
+      244 & 0x7F,
+      (244 >> 7) & 0x7F,
+    ];
+
+    serialPort.once("data", function(buffer) {
+      should.deepEqual(buffer, [
+        START_SYSEX,
+        SERIAL_MESSAGE,
+        SERIAL_REPLY | 0x08,
+        inBytes[0],
+        inBytes[1],
+        inBytes[2],
+        inBytes[3],
+        inBytes[4],
+        inBytes[5],
+        END_SYSEX]);
+      done();
+    });
+
+    serialPort.emit("data", [
+      START_SYSEX,
+      SERIAL_MESSAGE,
+      SERIAL_REPLY | 0x08,
+      inBytes[0],
+      inBytes[1],
+      inBytes[2],
+      inBytes[3],
+      inBytes[4],
+      inBytes[5],
+      END_SYSEX
+    ]);
+  });
+
+  it("should emit serial-data when serial data is received", function(done) {
```

Comment:
This however is a good test, maybe a multibyte version as well and then get rid of the 2 above tests that are really probably only testing MockSerialPort.